source/common/quic/quic_filter_manager_connection_impl.h

```diff
@@ -5,11 +5,14 @@
 #include "envoy/event/dispatcher.h"
 #include "envoy/network/connection.h"
 
+#define QUICHE_INCLUDE_1 "quiche/quic/core/quic_connection.h"
```

Comment:
What's this all about?  Can we have macro comments?

Reply:
Do we have a plan to get rid of these?  cc @DavidSchinazi yak shaver extrordinare

Reply:
What's the goal of this macro? What's wrong with having duplicated pragmas?

Reply:
2 things:
-QUICHE_INCLUDE_1 is the only one of these that is used.  We probably don't need 2 to 5
- The header include that needs the pragma seems to be only quiche/quic/core/quic_connection.h; can we add a wrapper header with appropriate pragmas and have that include be the one that is included through the rest of Envoy code?  Or have these pragmas added to quiche/quic/core/quic_connection.h via a quiche patch that Envoy adds while fetching the quiche dependency via bazel?

Reply:
> * QUICHE_INCLUDE_1 is the only one of these that is used.  We probably don't need 2 to 5

The plan is to use QUICHE_INCLUDE_X in other files as well, they includes more QUICHE files which also neeeds the pragma diagnose ignore.

> * The header include that needs the pragma seems to be only quiche/quic/core/quic_connection.h; can we add a wrapper header with appropriate pragmas and have that include be the one that is included through the rest of Envoy code?  Or have these pragmas added to quiche/quic/core/quic_connection.h via a quiche patch that Envoy adds while fetching the quiche dependency via bazel?

quic_connection.h is not the only one that needs the pragma, but most of them.

Reply:
I see what you're trying to do. I'd say just use the pragmas directly in places that need them for now (and remove quic_includes_ignores.h), and I can take an action item to fix QUICHE. The correct fix here is to make QUICHE build without warnings, not to disable them in Envoy.

Reply:
SG.